features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/TimelineView.kt

```diff
@@ -72,6 +74,7 @@ import kotlinx.coroutines.launch
 @Composable
 fun TimelineView(
     state: TimelineState,
+    timelineProtectionState: TimelineProtectionState,
```

Comment:
Maybe this could be used as a `CompositionLocalProvider`? This way we don't have it to explicitly pass it around through so many classes and functions, but it's ok as it is too.

Reply:
Add it to `TimelineRoomInfo ` as it's already passed everywhere?

Reply:
Not easy at the state is computed in MessagesPresenter as I need the state for the MessageComposer. I may revisit this later.